src/register/pmpcfgx.rs

```diff
@@ -1,7 +1,124 @@
 /// Physical memory protection configuration
+
 pub mod pmpcfg0 {
-    read_csr_as_usize!(0x3A0, __read_pmpcfg0);
-    write_csr_as_usize!(0x3A0, __write_pmpcfg0);
+    use bit_field::BitField;
+
+    #[derive(Clone, Copy, Debug)]
+    pub enum Permission {
+        NONE = 0,
+        R = 1,
+        W = 2,
+        RW = 3,
+        X = 4,
+        RX = 5,
+        WX = 6,
+        RWX = 7,
+    }
+
+    #[derive(Clone, Copy, Debug)]
+    pub enum Range {
+        OFF = 0,
+        TOR = 1,
+        NA4 = 2,
+        NAPOT = 3,
+    }
+
+    #[derive(Clone, Copy, Debug)]
+    pub struct Pmpconfig {
+        pub permission: Permission,
+        pub range_type: Range,
+        pub locked: bool,
+    }
+
+    #[derive(Clone, Copy, Debug)]
+    pub struct Pmpcfg0 {
+        bits: usize,
+    }
+
+    impl Pmpcfg0 {
+        ///Returns the pmp byte associated with the index
+        #[inline]
+        pub fn pmp_byte(&self, index: usize) -> usize {
```

Comment:
```suggestion
        pub fn pmp_byte(&self, index: usize) -> u8 {
```
Probably since it returns byte, it should have `u8` return type?